publish-frontend/docker/production/docker-compose.yml

```
@@ -0,0 +1,10 @@
+version: '3'
+
+services:
+  publish-frontend-production:
+    build:
+      context: ../../
+      dockerfile: docker/production/Dockerfile
+    image: publish-frontend-production
```

```suggestion
    image: publish-frontend-production
    environment:
      - NEXTAUTH_SECRET
```
> It seems I'm still missing something. I tested with all the variables in .env.local set, also Auth0 configured in the Strapi admin panel, but it's still giving the following giving the following error.

This is _probably_ the simplest solution to that issue. If you use this syntax, `docker compose up` will use the shell environment's value of `NEXTAUTH_SECRET` when it starts the containers. That way, you can start them with

```
NEXTAUTH_SECRET=whatever-you-want make start-production
```

---

I looked into this and the reason it's happening is because the environment file, `.env.local`, isn't being copied over before building the app. My thoughts on fixing this would be to use `.env` or `.env.production` file to store the values or we add another line below `COPY . .` to copy the environment file

---

I don't think we should, because that would bake secrets into the image. That's not completely out of the question, but it would require us to treat the images as secrets and that's a headache.

---

Gotcha. Thanks for the info. Your suggestion is better in that case.